File: app/src/main/java/com/kalap/contacts/CallLogsFragment.java
package com.kalap.contacts;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.kalap.contacts.adapters.CallLogAdapter;
import com.kalap.contacts.database.CallLogLoadListener;
import com.kalap.contacts.executors.CallLogExecutor;
import com.kalap.contacts.object.PhoneLog;

import java.util.ArrayList;

public class CallLogsFragment extends Fragment implements CallLogLoadListener {

    private RecyclerView callLogs;
    private ArrayList<PhoneLog> phoneLogs;

    public static CallLogsFragment newInstance() {

        Bundle args = new Bundle();

        CallLogsFragment fragment = new CallLogsFragment();
        fragment.setArguments(args);
        return fragment;
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.call_logs_fragment,container,false);
        callLogs = (RecyclerView) view.findViewById(R.id.call_logs);
        CallLogExecutor executor = new CallLogExecutor();
        executor.setListener(this);
        executor.loadCallLogs(getActivity());
        phoneLogs = new ArrayList<>();
        resetAdapter();
        return view;
    }

    private void resetAdapter() {
        getActivity().runOnUiThread(new Runnable() {
            @Override
            public void run() {
                CallLogAdapter callLogAdapter = new CallLogAdapter(getActivity(), phoneLogs);
                callLogs.setAdapter(callLogAdapter);
                callLogs.setLayoutManager(new LinearLayoutManager(getActivity()));
            }
        });
    }

    @Override
    public void onCallLogLoaded(final ArrayList<PhoneLog> phoneLogs) {
        this.phoneLogs = phoneLogs;
    }
}

File: app/src/main/java/com/kalap/contacts/CallLogsFragment.kt
package com.kalap.contacts

import android.content.pm.PackageManager
import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v7.widget.LinearLayoutManager
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.kalap.contacts.`object`.PhoneLog
import com.kalap.contacts.adapters.CallLogAdapter
import com.kalap.contacts.executors.CallLogExecutor
import com.kalap.contacts.listeners.CallLogLoadListener
import kotlinx.android.synthetic.main.call_logs_fragment.view.*
import java.util.*

class CallLogsFragment : Fragment(), CallLogLoadListener {

    private lateinit var phoneLogs: ArrayList<PhoneLog>
    private lateinit var root: View

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        root = inflater?.inflate(R.layout.call_logs_fragment, container, false)!!
        phoneLogs = ArrayList()
        resetAdapter()
        return root
    }

    override fun onResume() {
        super.onResume()
        val executor = CallLogExecutor(activity)
        executor.listener = this
        executor.loadCallLogs()
    }

    private fun resetAdapter() {
        activity.runOnUiThread {
            val callLogAdapter = CallLogAdapter(activity, phoneLogs)
            root.call_logs.adapter = callLogAdapter
            root.call_logs.layoutManager = LinearLayoutManager(activity)
        }
    }

    override fun onCallLogLoaded(phoneLogs: ArrayList<PhoneLog>) {
        this.phoneLogs = phoneLogs
        resetAdapter()
    }

    override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<String>, grantResults: IntArray) {
        super.onRequestPermissionsResult(requestCode, permissions, grantResults)
        when (requestCode) {
            103 -> {
                if (grantResults.isNotEmpty() && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
                    val executor = CallLogExecutor(activity)
                    executor.listener = this
                    executor.loadCallLogs()
                }
            }
        }
    }

    companion object {
        fun newInstance(): CallLogsFragment = CallLogsFragment()
    }
}
